Assets/MixedRealityToolkit/Services/MixedRealityToolkit.cs

```diff
@@ -244,30 +244,17 @@ public void ResetConfiguration(MixedRealityToolkitConfigurationProfile profile)
                 // Raycast provider reference is not managed by the MixedRealityToolkit class.
                 else if (typeof(IMixedRealitySpatialAwarenessSystem).IsAssignableFrom(interfaceType)) { spatialAwarenessSystem = null; }
                 else if (typeof(IMixedRealityTeleportSystem).IsAssignableFrom(interfaceType)) { teleportSystem = null; }
-
-                return true;
-            }
-
-            Tuple<Type, IMixedRealityService> registryInstance = new Tuple<Type, IMixedRealityService>(interfaceType, serviceInstance);
-
-            if (registeredMixedRealityServices.Contains(registryInstance))
-            {
-                registeredMixedRealityServices.Remove(registryInstance);
-                if (!(serviceInstance is IMixedRealityDataProvider))
-                {
-                    // Only remove IMixedRealityService or IMixedRealityExtensionService (not IMixedRealityDataProvider)
-                    MixedRealityServiceRegistry.RemoveService<T>(serviceInstance, this);
-                }
-                return true;
             }
 
-            return false;
+            return MixedRealityServiceRegistry.RemoveService<T>(serviceInstance, this);
         }
 
         /// <inheritdoc />
         public bool IsServiceRegistered<T>(string name = null) where T : IMixedRealityService
         {
-            return GetService<T>(name) != null;
+            T service;
```

Comment:
nit: consider checking for IMixedRealityDataProvider and log guidance to inquire with the appropriate service.